.github/workflows/check-pl-compat.yaml

```diff
@@ -38,7 +38,7 @@ jobs:
 
     - name: Install deps
       run: |
-        sudo apt-get install -y cmake ninja-build ccache libomp-dev libasan6
+        sudo apt-get install -y cmake ninja-build ccache libomp-dev liblapack-dev libasan6
```

Comment:
If we're adding the package here it probably needs to be added to all our other actions as well (where appropriate).
Our install docs also need to reflect this new dependency:
https://github.com/PennyLaneAI/catalyst/blob/300ff893e3d75758a2fb08f48ef8ed5f81099a36/doc/dev/installation.rst?plain=1#L66